### PR TITLE
Glutin port refactoring

### PR DESCRIPTION
**Pull request by paulrouget**
Glutin port refactoring in preparation for the compositor and libservo refactoring.

In theory, the only behavior change is for headless mode. The headless event loop now uses winit's event loop (but still headless).

Notes:
- headless and glutin window implementations are now separated
- I split the methods of the embedder in 2: window specific and general methods. In the future, we still want the app to run even without a window or with multiple windows

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23233)

### CODE REVIEW

**Comment by atouchet** [comment]
Would you be able to update to the new glutin 0.21 release along with this?

**Comment by paulrouget** [comment]
> Would you be able to update to the new glutin 0.21 release along with this?

I might do it before this PR. Not sure yet.

**Comment by paulrouget** [comment]
the winit event loop on Linux can't work in headless mode :(

**Comment by paulrouget** [comment]
Broke multiprocess.

**Comment by paulrouget** [comment]
Note to self: we should check if the tests run any faster/slower than before.

**Comment by paulrouget** [comment]
This should be ready for review.

File: ports/glutin/embedder.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    proxy: self.proxy.clone(),
+                })
+            }
+        }
+
+        Box::new(GlutinEventLoopWaker::new(&self.events_loop.borrow()))
+    }
+
+    fn register_vr_services(&self, services: &mut VRServiceManager, heartbeats: &mut Vec<Box<WebVRMainThreadHeartbeat>>) {
+        if opts::get().headless {
```
Comment (jdm) [inline ports/glutin/embedder.rs]: This looks backwards.

**Review by jdm — APPROVED** [review]
This looks like a really nice improvement!

File: ports/glutin/embedder.rs (hunk 1 above)
Comment (paulrouget) [reply ports/glutin/embedder.rs]: Fixed.

**Comment by paulrouget** [comment]
Squashed and rebased.

**Pull request merged**